### PR TITLE
[feature](security) Support block specific query with AST names

### PR DESCRIPTION
**Pull request by Hastyshell**
### What problem does this PR solve?

Support block specific query with AST names when necessary for security reasons, configure the name list in fe.conf, for example:

```
block_sql_ast_names="CreateFileStmt, CreateFunctionStmt"
```

### Release note

Support to block some SQL statements, e.g. CreateFileStmt, when necessary for security reasons.

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [x] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -3224,4 +3224,9 @@ public static int metaServiceRpcRetryTimes() {
             "For testing purposes, all queries are forcibly forwarded to the master to verify"
                     + "the behavior of forwarding queries."})
     public static boolean force_forward_all_queries = false;
+
+    @ConfField(description = {"用于禁用某些SQL，配置项为AST的class simple name列表，用逗号间隔开",
+            "For disabling certain SQL queries, the configuration item is a list of simple class names of AST,"
+                    + " separated by commas."})
+    public static String block_sql_ast_names = "";
```
Comment (gavinchou) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:3231]: forbiden_stmts

**Comment by TangSiyang2001** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/parser/NereidsParserTest.java [hunk 2]
```diff
@@ -704,4 +706,41 @@ public void testQualify() {
             nereidsParser.parseSingle(sql);
         }
     }
+
+    @Test
+    public void testBlockSqlAst() {
```
Comment (gavinchou) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/parser/NereidsParserTest.java:711]: test `create file`, `create function` statment too

File: fe/fe-core/src/test/java/org/apache/doris/qe/StmtExecutorTest.java [hunk 3]
```diff
@@ -801,4 +802,59 @@ public void testUseWithCatalogFail(@Mocked UseStmt useStmt, @Mocked SqlParser pa
 
         Assert.assertEquals(QueryState.MysqlStateType.ERR, state.getStateType());
     }
+
+    @Test
+    public void testBlockSqlAst(@Mocked UseStmt useStmt, @Mocked SqlParser parser) throws Exception {
```
Comment (gavinchou) [inline fe/fe-core/src/test/java/org/apache/doris/qe/StmtExecutorTest.java:807]: test `create file`, `create function` statment too

also test some commands that should not be blocked

**Comment by TangSiyang2001** [comment]
run feut

**Comment by TangSiyang2001** [comment]
run feut

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Comment by TangSiyang2001** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by Yukang-Lian — APPROVED** [review]

**Review by gavinchou — APPROVED** [review]

**Review by morningman — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**